driver-core/src/test/functional/com/mongodb/operation/CreateCollectionOperationSpecification.groovy

```diff
@@ -185,6 +185,46 @@ class CreateCollectionOperationSpecification extends OperationFunctionalSpecific
         }
     }
 
+    def 'should create collection with autoIndex'() {
+        given:
+        assert !collectionNameExists(getCollectionName())
+
+        when:
+        new CreateCollectionOperation(getDatabaseName(), getCollectionName())
+                .autoIndex(true)
+                .execute(getBinding())
+
+        then:
+        collectionNameExists(getCollectionName())
+
+        when:
```

Comment:
Couple of suggestions for the tests:

1. Combine the two tests into one using Spock's where syntax: e.g.

    where:
    autoIndex  | expectedNumberOfIndexes
    true            | 1
    false           | 0

2. Combine the two "then" clauses.  The "when" clause is for the code that is under test, so the second one is a mis-use of it, and can be folded into the "then" clause